opm/utility/ECLWellSolution.cpp

```diff
@@ -224,105 +153,41 @@ namespace Opm
-                SubSelectLGRBlock subselect(restart_.get(), lgr_index);
-                {
-                    // Read well data for LGR grid.
-                    std::vector<WellData> wd = readWellData(grid_index);
-                    // Append to set of all well data.
-                    all_wd.insert(all_wd.end(), wd.begin(), wd.end());
-                }
-            }
-            return all_wd;
-        }
-    }
-
-
-
-
-    ecl_kw_type*
-    ECLWellSolution::getKeyword(const std::string& fieldname) const
-    {
-        const int local_occurrence = 0; // This should be correct for all the well-related keywords.
-        if (ecl_file_get_num_named_kw(restart_.get(), fieldname.c_str()) == 0) {
-            throw std::runtime_error("Could not find field " + fieldname);
-        }
-        return ecl_file_iget_named_kw(restart_.get(), fieldname.c_str(), local_occurrence);
-    }
-
-
-
-
-    std::vector<double>
-    ECLWellSolution::loadDoubleField(const std::string& fieldname) const
-    {
-        ecl_kw_type* keyword = getKeyword(fieldname);
-        std::vector<double> field_data;
-        field_data.resize(ecl_kw_get_size(keyword));
-        ecl_kw_get_data_as_double(keyword, field_data.data());
-        return field_data;
-    }
-
-
-
-
-    std::vector<int>
-    ECLWellSolution::loadIntField(const std::string& fieldname) const
-    {
-        ecl_kw_type* keyword = getKeyword(fieldname);
-        std::vector<int> field_data;
-        field_data.resize(ecl_kw_get_size(keyword));
-        ecl_kw_get_memcpy_int_data(keyword, field_data.data());
-        return field_data;
-    }
-
-
-
-
-    std::vector<std::string>
-    ECLWellSolution::loadStringField(const std::string& fieldname) const
-    {
-        ecl_kw_type* keyword = getKeyword(fieldname);
-        std::vector<std::string> field_data;
-        const int size = ecl_kw_get_size(keyword);
-        field_data.resize(size);
-        for (int pos = 0; pos < size; ++pos) {
-            field_data[pos] = ecl_kw_iget_char_ptr(keyword, pos);
+        // Read well data for global grid.
+        std::vector<WellData> all_wd{};
+        for (int grid_index = 0; grid_index < num_grids; ++grid_index) {
+            std::vector<WellData> wd = readWellData(restart, grid_index);
+            // Append to set of all well data.
+            all_wd.insert(all_wd.end(), wd.begin(), wd.end());
         }
-        return field_data;
+        return all_wd;
     }
 
 
 
 
     std::vector<ECLWellSolution::WellData>
-    ECLWellSolution::readWellData(const int grid_index) const
+    ECLWellSolution::readWellData(const ECLResultData& restart, const int grid_index) const
     {
         // Note: this function is expected to be called in a context
-        // where the correct restart block and grid subblock has already
-        // been selected using the ert block mechanisms.
+        // where the correct restart block using the ert block mechanisms.
```

Comment:
Strange comment after change, I guess you mean that `restart` should be viewing the correct restart block?